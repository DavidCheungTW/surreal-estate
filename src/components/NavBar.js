import React from "react";
import { Link } from "react-router-dom";
import logo from "../images/logo.png";
import "../styles/navbar.css";

const NavBar = ({ menu }) => {
  let classVP;
  let classAP;

  if (menu === "classVP") {
    classVP = `navbar-links-item ${menu}`;
    classAP = "navbar-links-item";
  } else {
    classVP = "navbar-links-item";
    classAP = `navbar-links-item ${menu}`;
  }

  return (
    <div className="navbar">
      <img src={logo} className="navbar-logo" alt="logo" />
      <ul className="navbar-links">
        <li className={classVP}>
          <Link className="item" to="/">
            View Properties
          </Link>
        </li>
        <li className={classAP}>
          <Link className="item" to="add-property">
            Add a Property
          </Link>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
