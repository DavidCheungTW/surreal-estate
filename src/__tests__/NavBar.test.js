import React from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import { render, screen } from "@testing-library/react";

import renderer from "react-test-renderer";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  xit("Renders as expected", () => {
    const rendered = renderer.create(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    expect(rendered).toMatchSnapshot();
  });

  xit("renders learn react link", () => {
    const history = createMemoryHistory();
    render(
      <Router history={history}>
        <NavBar />
      </Router>
    );

    // expect(screen.getByText("View Properties")).toBeTruthy();
    // expect(screen.getByText("View Properties")).toBeInstanceOf(
    //   HTMLParagraphElement
    // );
    // expect(screen.getByText("Add a Property")).toBeTruthy();
    // expect(screen.getByText("Add a Property")).toBeInstanceOf(
    //   HTMLParagraphElement
    // );
  });
});
