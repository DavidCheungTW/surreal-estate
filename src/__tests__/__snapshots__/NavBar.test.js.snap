// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Renders as expected 1`] = `
<div
  className="navbar"
>
  <img
    alt="logo"
    className="navbar-logo"
    src="logo.png"
  />
  <ul
    className="navbar-links"
  >
    <li
      className="navbar-links-item"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <a
          href="/"
        >
          View Properties
        </a>
      </a>
    </li>
    <li
      className="navbar-links-item"
    >
      <a
        href="/add-property"
        onClick={[Function]}
      >
        <a
          href="add-property"
        >
          Add a Property
        </a>
      </a>
    </li>
  </ul>
</div>
`;
